pinecone/provider/collection_resource.go

```diff
@@ -179,30 +182,30 @@ func (r *CollectionResource) Delete(ctx context.Context, req resource.DeleteRequ
 		return
 	}
 
-	err := r.client.DeleteCollection(ctx, data.Name.ValueString())
+	err := r.client.DeleteCollection(ctx, data.Id.ValueString())
 	if err != nil {
 		resp.Diagnostics.AddError("Failed to delete collection", err.Error())
 		return
 	}
 	// Wait for collection to be deleted
-	// Create() is passed a default timeout to use if no value
+	// Delete() is passed a default timeout to use if no value
 	// has been supplied in the Terraform configuration.
-	deleteTimeout, diags := data.Timeouts.Delete(ctx, defaultIndexDeleteTimeout)
+	deleteTimeout, diags := data.Timeouts.Delete(ctx, defaultCollectionDeleteTimeout)
 	resp.Diagnostics.Append(diags...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
 	err = retry.RetryContext(ctx, deleteTimeout, func() *retry.RetryError {
-		collection, err := r.client.DescribeCollection(ctx, data.Id.ValueString())
-		// tflog.Info(ctx, fmt.Sprintf("Deleting Collection. Status: '%s'", collection.Status))
```

Comment:
This may have been commented out because previously we were trying to access `collection` in the case of errors.